packages/parameters/src/ssm/SSMProvider.ts

```diff
@@ -144,13 +144,12 @@ class SSMProvider extends BaseProvider {
     options?: SSMGetOptionsInterface
   ): Promise<string | undefined> {
     const sdkOptions: GetParameterCommandInput = {
+      ...(options?.sdkOptions || {}),
       Name: name,
     };
     if (options) {
-      if (options.hasOwnProperty('decrypt')) sdkOptions.WithDecryption = options.decrypt;
-      if (options.hasOwnProperty('sdkOptions')) {
-        Object.assign(sdkOptions, options.sdkOptions);
-      }
+      if (options.hasOwnProperty('decrypt'))
+        sdkOptions.WithDecryption = options.decrypt;
```

Comment:
Now that we have types, ca we reduce the if blocks to a more simpler version?
```ts
sdk.WithDecryption = options?.decrypt? : options.decrypt : undefined;
```

Reply:
I'm not sure this can work.

This is the current implementation:
```ts
const sdkOptions: GetParameterCommandInput = {
  ...(options?.sdkOptions || {}),
  Name: name,
};
if (options) {
  if (options.hasOwnProperty('decrypt'))
    sdkOptions.WithDecryption = options.decrypt;
  }
}
```

In this method, users can influence the behavior around decrypt in two ways:
- By using the `decrypt` option (i.e. `provider.get('something', { decrypt: true });`)
- By passing the `WithDecryption` parameter directly to the SDK (i.e. `provider.get('something', { sdkOptions: { WithDecryption: true }});`)

The former takes precedence over the second.

If we were to modify the statement like suggested, we'd always overwrite the option either with the value of `options.decrypt` or with `undefined`. This would mean never considering the value of `options.sdkOptions` when one is passed.

Now, we could make the ternary operator statement include this case and do something like this:
```ts
sdk.WithDecryption = options?.decrypt? : options.decrypt : options?.sdkOptions?.WithDecryption ? options.sdkOptions.WithDecryption : undefined;
```

but I think this, while shorter in terms of LOC it's less readable. Generally speaking I'm not too concerned about LOC since all this is transpiled to JS before being published.

Thoughts?

Reply:
I went with this one:
```ts
sdk.WithDecryption = options?.decrypt !== undefined ? options.decrypt : sdk.WithDecryption
```

this way, if the user passes `options.decrypt === false` we still respect their value. Not passing it would have evaluated to `false`, and thus `undefined`. For most cases the two are equal but in some cases it might need to exactly be `false`, so I made the first comparison stricter.